### ISSUE TITLE
Group by category for hist

### ISSUE THREAD

**Issue by papajohn** [data-8/datascience#280]
It's a pain to draw two overlaid histograms that differ by a categorical variable. E.g., to overlay these two:

```
nonsmokers = baby.where('Maternal Smoker', are.equal_to(False))
baby.where('Maternal Smoker', False).hist('Birth Weight')
baby.where('Maternal Smoker', True).hist('Birth Weight')
```

the simplest expression is

```
baby.where('Maternal Smoker', False).bin('Birth Weight')\
  .join('bin', baby.where('Maternal Smoker', True).bin('Birth Weight')).\
  relabeled(1, 'False').relabeled(2, 'True').hist(bin_column='bin')
```

**Comment by papajohn** [comment]
Proposed syntax

`baby.hist('Birth Weight', group='Maternal Smoker')`

**Comment by a-adhikari** [comment]
Great! Agree that it's needed. Could also be used to overlay histograms using data from two different tables, e.g. distribution of incomes from Table 1 and distribution of incomes from Table 2.
Ani

--
Ani Adhikari
Teaching Professor
Department of Statistics
University of California, Berkeley

**Comment by henryem** [comment]
@papajohn I'd like to use this, and I'd be happy to implement it.

**Comment by papajohn** [comment]
Great!

**Comment by henryem** [comment]
Actually @papajohn, I happened to find `pivot_hist` just now.  Is this the point of `pivot_hist`?  The issue is that `pivot_hist` really makes a bar graph, with each bin getting side-by-side bars for each category, instead of overlaying the histograms.  (I guess that's why it's called `pivot_hist` rather than `group_hist`, since it thinks of the bins as categories.)

It's confusing to have both, and I wonder if the behavior of `pivot_hist` really makes sense.  Do you know if it's used?  We could have an option to recover the `pivot_hist` behavior, like `t.hist('bar', bins='foo', side_by_side=True)`.

**Comment by papajohn** [comment]
As far as I know, pivot_hist is not used. I agreed that it's an unusual
thing and doesn't fit nicely into the rest of the module.

**Comment by henryem** [comment]
Closed by #293.
